Reopen comment model

mattegan111 opened the pull request:
We accidentally closed comment-model branch and cannot reopen it because post-model branch, which was the base, has been deleted.

Here I've implemented changes from #5, the original branch.

@nathanrobb you might need to jump between PR's for context, sorry!

mattegan111 committed:
Rename model key

Rename key because the db entry will contain an Id rather than the object itself.

mattegan111 committed:
Added missing ref

This way we can store a reference to another model so that at some point we can populate those to get the complete model by that reference.

mattegan111 committed:
Fix validation

The 'post' value is expected to be part of the request, not a user input, so we can replace the validation with an if statement and error message.

mattegan111 committed:
Separate routes for adding/removing likes

We can adhere to a RESTful structure by having separate routes for adding a like to a post and deleting a like from a post - as opposed to an 'edit post' function that handles both of these as well as editing the text of a post.

mattegan111 committed:
Refactor find

use different find methods to reduce code.

nathanrobb commented on server/routes/comments.js:
```diff
@@ -105,63 +95,98 @@ router.put(
+				.json({ msg: 'Comment not found' });
+		}
+
+		if (comment.likes.includes(ObjectId(req.user.id))) {
+			res.json({ msg: 'You already like this comment' });
+		}
```
Were we wanting to `return` here so we don't execute the rest of the methods code?

nathanrobb reviewed (approved):
LGTM

mattegan111 committed:
Fix res to return inside if statement

res should be returned and function should end when calling res.json() inside if statement.
